### DISCUSSION TITLE
Embedding a map with default values doesn't work when using an object

### DISCUSSION THREAD

**Discussion by harpratap** [cue-lang/cue#1365] [Q&A]
### What version of CUE are you using (`cue version`)?

<pre>
$ cue version
cue version 0.4.0 darwin/amd64
</pre>

### Does this issue reproduce with the latest release?

Yes

### What did you do?

Trying to embed a definition inside another one and extending it with additional field called `tags`

```
#Metadata: {
    name: string
}

#MyDef: {
    metadata: {
        #Metadata
        tags: [string]: string
        tags: "hello": "world" // this tag should always exist
    }
}

meta: #Metadata & {
    name: "Harry"
}

out: #MyDef & {
    metadata: {
        meta
        // only works if you add a tag here
        // tags: "abc": "xyz"
    }
}
```
It works when I change from using an object `meta` to directly defining `name` inside `out.metadata`

`out: #MyDef & {metadata: name: "Harry"}`

### What did you expect to see?

```
$ cue export test.cue
{
    "meta": {
        "name": "Harry"
    },
    "out": {
        "metadata": {
            "name": "Harry",
            "tags": {
                "hello": "world"
            }
        }
    }
}
```

### What did you see instead?

```
out.metadata: field not allowed: tags:
    ./test.cue:1:12
    ./test.cue:7:9
    ./test.cue:9:9
    ./test.cue:10:9
    ./test.cue:14:7
    ./test.cue:14:19
    ./test.cue:18:6
    ./test.cue:19:15
    ./test.cue:20:9
```

**Comment by verdverm** [comment]
I suspect this is the expected behavior of an embedding based on: https://cuelang.org/docs/references/spec/#embedding

https://github.com/cue-lang/cue/discussions/1090 may also be helpful or related

What is happening is when you embed meta, which is closed, that is closing the RHS of the unification, and then the tags from `#MyDef` are no longer allowed.

There are a few different variations that will work:

- [Avoid schema with meta](https://cuelang.org/play/?id=WQO8w2J-TSs#cue@export@cue)
- [Open #Metadata](https://cuelang.org/play/?id=xfbMDqa7W4H#cue@export@cue)
- [Embed into and open meta](https://cuelang.org/play/?id=50AEVREnKGo#cue@export@cue)
- [Use field comprehension](https://cuelang.org/play/?id=Ah7lqOWffpc#cue@export@cue)

Generally, I would avoid embedding data and instead unify. In other words, change

```cue
    metadata: {
         meta
```

into

```cue
    metadata: meta & {
```

I believe most of the alternatives above work either. I personally use embeddings for building up schemas and then use conjunctions (`&`) to set concrete values, even if only partially specifying. I think it is valid to say embeddings redefine shape.

The field comprehension variation is probably the strictest version and closest to what you are after.

---

On a side note, this may be better as a GitHub Discussion

**Comment by harpratap** [comment]
@verdverm Oh I see, thanks for the explanation. 

> On a side note, this may be better as a GitHub Discussion

Sorry about that. Looks like I cannot convert issues to discussions but only the other way around.

**Reply by myitcv** [reply]
> Looks like I cannot convert issues to discussions but only the other way around.

Done!
